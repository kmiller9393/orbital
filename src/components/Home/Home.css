.review-link {
  text-decoration: none;
}

.home-button {
  background-color: aqua;
}
