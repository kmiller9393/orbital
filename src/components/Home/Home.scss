.main-header-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 150px;
  justify-content: space-between;
  margin-top: 1.5rem;
}

.header-title {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  font-size: 1.6rem;
  font-weight: lighter;
}

.main-header-image {
  height: 55px;
  width: 125px;
}

.review-link {
  color: #f7f7f7;
  font-size: 1.5rem;
  text-decoration: none;
}

.homepage-buttons {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 4rem;
}

.home-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  height: 45px;
  margin-top: 1.5rem;
  width: 200px;
}
