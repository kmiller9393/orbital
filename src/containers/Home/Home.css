.main-header-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 150px;
  justify-content: space-between;
  padding-top: 1.5rem; }

.header-title {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  font-size: 1.6rem;
  font-weight: lighter; }

.main-header-image {
  height: 55px;
  width: 125px; }

.review-link-1,
.trivia-link-1,
.current-link-1 {
  color: #f7f7f7;
  font-size: 1.5rem;
  text-decoration: none; }

.homepage-buttons {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 3rem; }

.home-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  cursor: pointer;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  height: 55px;
  margin-top: 2.5rem;
  width: 220px; }
