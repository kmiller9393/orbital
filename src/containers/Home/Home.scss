.main-header-container {
  align-items: center;
  display: flex;
  flex-direction: column;
  height: 150px;
  justify-content: space-between;
  padding-top: 1.5rem;
}

.header-title {
  color: #f7f7f7;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  font-size: 1.6rem;
  font-weight: lighter;
  text-shadow: 2px 4px 3px rgba(0, 0, 0, 0.7);
}

.main-header-image {
  height: 55px;
  width: 125px;
}

.review-link-1,
.trivia-link-1,
.current-link-1 {
  color: #f7f7f7;
  font-size: 1.5rem;
  text-decoration: none;
}

.homepage-buttons {
  align-items: center;
  display: flex;
  flex-direction: column;
  margin-top: 3rem;
}

.home-button {
  background-color: #009c92;
  border-color: #009c92;
  border-radius: 5px;
  box-shadow: 0px 8px 15px rgba(0, 0, 0, 0.5);
  cursor: pointer;
  font-family: 'Chalkboard SE', 'Times New Roman', serif;
  height: 55px;
  margin-top: 2.5rem;
  text-shadow: 2px 4px 3px rgba(0, 0, 0, 0.3);
  width: 220px;
}

.home-button:active {
  background-color: #02b0a4;
}
