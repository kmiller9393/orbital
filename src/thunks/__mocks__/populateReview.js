export const populateReview = jest.fn();
