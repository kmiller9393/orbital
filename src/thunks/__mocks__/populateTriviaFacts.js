export const populateTriviaFacts = jest.fn();
