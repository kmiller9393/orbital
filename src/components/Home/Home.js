import React, { Component } from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { populateTriviaFacts } from '../../thunks';
import { populateReview } from '../../thunks';
import orbital from '../../images/orbital-header.svg';
import './Home.css';

class Home extends Component {
  setTriviaItems = async () => {
    const { populateTrivia } = this.props;
    await populateTrivia();
  };

  setReviewItems = async () => {
    const { populateReview } = this.props;
    await populateReview();
  };

  render() {
    return (
      <div>
        <header className="main-header-container">
          <img className="main-header-image" src={orbital} alt="Orbital" />
          <h1 className="header-title">Welcome Back, Brandon</h1>
        </header>
        <div className="homepage-buttons">
          <button className="home-button">
            <NavLink
              className="review-link"
              to="/review"
              onClick={this.setReviewItems}
            >
              Review
            </NavLink>
          </button>
          <button className="home-button">
            <NavLink
              className="review-link"
              to="/trivia-center"
              onClick={this.setTriviaItems}
            >
              Trivia Center
            </NavLink>
          </button>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  populateTrivia: () => dispatch(populateTriviaFacts()),
  populateReview: () => dispatch(populateReview())
});

export default connect(
  null,
  mapDispatchToProps
)(Home);
