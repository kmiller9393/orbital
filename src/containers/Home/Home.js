import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { populateTriviaFacts } from '../../thunks/populateTriviaFacts';
import { populateReview } from '../../thunks/populateReview';
import { populateEvents } from '../../thunks/populateEvents';
import orbital from '../../images/orbital-header.svg';
import './Home.css';

export class Home extends Component {
  handleTriviaRoute = async () => {
    const { populateTrivia, history } = this.props;

    history.push('/trivia-center');

    await populateTrivia();
  };

  handleReviewRoute = async () => {
    const { populateReview, history } = this.props;

    history.push('/review');

    await populateReview();
  };

  handleEventsRoute = async () => {
    const { populateEvents, history } = this.props;

    history.push('/current-events');

    await populateEvents();
  };

  render() {
    return (
      <div>
        <header className="main-header-container">
          <img className="main-header-image" src={orbital} alt="Orbital" />
          <h1 className="header-title">Welcome Back</h1>
        </header>
        <div className="homepage-buttons">
          <button
            className="home-button review-link-1"
            onClick={this.handleReviewRoute}
          >
            Review
          </button>
          <button
            className="home-button trivia-link-1"
            onClick={this.handleTriviaRoute}
          >
            Trivia Center
          </button>
          <button
            className="home-button current-link-1"
            onClick={this.handleEventsRoute}
          >
            Current Events
          </button>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = dispatch => ({
  populateTrivia: () => dispatch(populateTriviaFacts()),
  populateReview: () => dispatch(populateReview()),
  populateEvents: () => dispatch(populateEvents())
});

export default connect(
  null,
  mapDispatchToProps
)(Home);

const { func } = PropTypes;
Home.propTypes = {
  populateTrivia: func,
  populateReview: func
};
