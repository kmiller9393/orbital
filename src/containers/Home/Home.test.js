import React from 'react';
import { shallow } from 'enzyme';
import { Home, mapDispatchToProps } from './Home';
import { populateTriviaFacts } from '../../thunks/populateTriviaFacts';
import { populateReview } from '../../thunks/populateReview';

jest.mock('../../thunks/populateTriviaFacts');
jest.mock('../../thunks/populateReview');

describe('Home', () => {
  let wrapper;
  let mockPopulateTrivia;
  let mockPopulateReview;

  beforeEach(() => {
    mockPopulateTrivia = jest.fn();
    mockPopulateReview = jest.fn();
    wrapper = shallow(
      <Home
        populateTrivia={mockPopulateTrivia}
        populateReview={mockPopulateReview}
      />
    );
  });

  describe('Home component', () => {
    it('should match the snapshot', () => {
      expect(wrapper).toMatchSnapshot();
    });

    it('should call setTriviaItems when the Trivia Center button is clicked', () => {
      const spy = jest.spyOn(wrapper.instance(), 'setTriviaItems');
      wrapper.instance().forceUpdate();
      wrapper.find('[name="trivia-link"]').simulate('click');
      expect(spy).toHaveBeenCalled();
    });

    it('should call setReviewItems when the Review button is clicked', () => {
      const spy = jest.spyOn(wrapper.instance(), 'setReviewItems');
      wrapper.instance().forceUpdate();
      wrapper.find('[name="review-link"]').simulate('click');
      expect(spy).toHaveBeenCalled();
    });
  });

  describe('mapDispatchToProps', () => {
    it('should call dispatch when using a function from mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = populateTriviaFacts();

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.populateTrivia();

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });

    it('should call dispatch when using a function from mapDispatchToProps', () => {
      const mockDispatch = jest.fn();
      const actionToDispatch = populateReview();

      const mappedProps = mapDispatchToProps(mockDispatch);
      mappedProps.populateReview();

      expect(mockDispatch).toHaveBeenCalledWith(actionToDispatch);
    });
  });
});
