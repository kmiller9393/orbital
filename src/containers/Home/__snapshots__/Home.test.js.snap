// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home Home component should match the snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Home />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <header
          className="main-header-container"
>
          <img
                    alt="Orbital"
                    className="main-header-image"
                    src="orbital-header.svg"
          />
          <h1
                    className="header-title"
          >
                    Welcome Back
          </h1>
</header>,
        <div
          className="homepage-buttons"
>
          <button
                    className="home-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="review-link"
                              onClick={[Function]}
                              to="/review"
                    >
                              Review
                    </NavLink>
          </button>
          <button
                    className="home-button"
          >
                    <NavLink
                              activeClassName="active"
                              aria-current="page"
                              className="review-link"
                              onClick={[Function]}
                              to="/trivia-center"
                    >
                              Trivia Center
                    </NavLink>
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <img
              alt="Orbital"
              className="main-header-image"
              src="orbital-header.svg"
/>,
            <h1
              className="header-title"
>
              Welcome Back
</h1>,
          ],
          "className": "main-header-container",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "alt": "Orbital",
              "className": "main-header-image",
              "src": "orbital-header.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Welcome Back",
              "className": "header-title",
            },
            "ref": null,
            "rendered": "Welcome Back",
            "type": "h1",
          },
        ],
        "type": "header",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              className="home-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="review-link"
                            onClick={[Function]}
                            to="/review"
              >
                            Review
              </NavLink>
</button>,
            <button
              className="home-button"
>
              <NavLink
                            activeClassName="active"
                            aria-current="page"
                            className="review-link"
                            onClick={[Function]}
                            to="/trivia-center"
              >
                            Trivia Center
              </NavLink>
</button>,
          ],
          "className": "homepage-buttons",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="review-link"
                onClick={[Function]}
                to="/review"
>
                Review
</NavLink>,
              "className": "home-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Review",
                "className": "review-link",
                "onClick": [Function],
                "to": "/review",
              },
              "ref": null,
              "rendered": "Review",
              "type": [Function],
            },
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <NavLink
                activeClassName="active"
                aria-current="page"
                className="review-link"
                onClick={[Function]}
                to="/trivia-center"
>
                Trivia Center
</NavLink>,
              "className": "home-button",
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "aria-current": "page",
                "children": "Trivia Center",
                "className": "review-link",
                "onClick": [Function],
                "to": "/trivia-center",
              },
              "ref": null,
              "rendered": "Trivia Center",
              "type": [Function],
            },
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <header
            className="main-header-container"
>
            <img
                        alt="Orbital"
                        className="main-header-image"
                        src="orbital-header.svg"
            />
            <h1
                        className="header-title"
            >
                        Welcome Back
            </h1>
</header>,
          <div
            className="homepage-buttons"
>
            <button
                        className="home-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="review-link"
                                    onClick={[Function]}
                                    to="/review"
                        >
                                    Review
                        </NavLink>
            </button>
            <button
                        className="home-button"
            >
                        <NavLink
                                    activeClassName="active"
                                    aria-current="page"
                                    className="review-link"
                                    onClick={[Function]}
                                    to="/trivia-center"
                        >
                                    Trivia Center
                        </NavLink>
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <img
                alt="Orbital"
                className="main-header-image"
                src="orbital-header.svg"
/>,
              <h1
                className="header-title"
>
                Welcome Back
</h1>,
            ],
            "className": "main-header-container",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "alt": "Orbital",
                "className": "main-header-image",
                "src": "orbital-header.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Welcome Back",
                "className": "header-title",
              },
              "ref": null,
              "rendered": "Welcome Back",
              "type": "h1",
            },
          ],
          "type": "header",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                className="home-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="review-link"
                                onClick={[Function]}
                                to="/review"
                >
                                Review
                </NavLink>
</button>,
              <button
                className="home-button"
>
                <NavLink
                                activeClassName="active"
                                aria-current="page"
                                className="review-link"
                                onClick={[Function]}
                                to="/trivia-center"
                >
                                Trivia Center
                </NavLink>
</button>,
            ],
            "className": "homepage-buttons",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="review-link"
                  onClick={[Function]}
                  to="/review"
>
                  Review
</NavLink>,
                "className": "home-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Review",
                  "className": "review-link",
                  "onClick": [Function],
                  "to": "/review",
                },
                "ref": null,
                "rendered": "Review",
                "type": [Function],
              },
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": <NavLink
                  activeClassName="active"
                  aria-current="page"
                  className="review-link"
                  onClick={[Function]}
                  to="/trivia-center"
>
                  Trivia Center
</NavLink>,
                "className": "home-button",
              },
              "ref": null,
              "rendered": Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "activeClassName": "active",
                  "aria-current": "page",
                  "children": "Trivia Center",
                  "className": "review-link",
                  "onClick": [Function],
                  "to": "/trivia-center",
                },
                "ref": null,
                "rendered": "Trivia Center",
                "type": [Function],
              },
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "attachTo": undefined,
    "hydrateIn": undefined,
  },
}
`;
